### PR TITLE
Past Events Functionality and Edit/Delete owned events functionality

### PR DESCRIPTION
**Pull request by Tcintra**
This pull request incorporates commits from the previous pull request that had merge conflicts, and new functionality for editing and deleting owned events.

### CODE REVIEW

**Review by christinachance — APPROVED** [review]

File: project/src/main/java/com/google/sps/servlets/EditEventServlet.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    String date = request.getParameter("edit-date");
+    String time = request.getParameter("edit-time");
+    // Format: hh:mm XM
+    String hours = time.substring(0, 2);
+    String minutes = time.substring(3, 5);
+    SimpleDateFormat formatter = new SimpleDateFormat("MMM dd, yyyy");
+    long dateTimestamp = 0;
+    try {
+      Date formattedDate = formatter.parse(date);
+      long milliseconds = Long.parseLong(hours)*3600000 + Long.parseLong(minutes)*60000;
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/EditEventServlet.java:63]: add spaces around  `*`

File: project/src/main/java/com/google/sps/servlets/EditEventServlet.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      event.setProperty("description", description);
+      event.setProperty("dateTimestamp", dateTimestamp);
+ 
+      datastore.put(event);
+      response.sendRedirect("/events.html");
+    } catch(EntityNotFoundException e) {
+      return;
+    }
+  }
+}
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/EditEventServlet.java:98]: one more line

File: project/src/main/java/com/google/sps/servlets/EditEventServlet.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      event.setProperty("time", time);
+      event.setProperty("type", type);
+      event.setProperty("attendance", attendance);
+      event.setProperty("description", description);
+      event.setProperty("dateTimestamp", dateTimestamp);
+ 
+      datastore.put(event);
+      response.sendRedirect("/events.html");
+    } catch(EntityNotFoundException e) {
+      return;
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/EditEventServlet.java:95]: maybe log the error message before return for debugging purpose?

File: project/src/main/java/com/google/sps/servlets/LoadExploreServlet.java [hunk 4]
```diff
@@ -26,48 +26,62 @@
...
+            String name = (String) entity.getProperty("name");
+            String location = (String) entity.getProperty("location");
+            String date = (String) entity.getProperty("date");
+            String time = (String) entity.getProperty("time");
+            String description = (String) entity.getProperty("description");
+            String type = (String) entity.getProperty("attendance");
+            String attendance = (String) entity.getProperty("type");
+            long timestamp = (long) entity.getProperty("timestamp");
+ 
+            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, entity.getProperty("email").equals(email), "ApprovedEvent");
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/LoadExploreServlet.java:77]: extract `entity.getProperty("email").equals(email)` to a var for easy understanding.

File: project/src/main/java/com/google/sps/servlets/LoadPastEventsServlet.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      Gson gson = new Gson();
+
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(pastEvents));
+    }
+    else {
+      response.sendRedirect("/login.html");
+    }
+  }
+}
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/LoadPastEventsServlet.java:92]: one more line

**Review by wulian1223 — APPROVED** [review]
LGTM after applying comments.

**Review by rivera-fernando — APPROVED** [review]

**Pull request merged**